SdarotAPI/SdarotDriver.cs

```diff
@@ -58,48 +71,32 @@ public async Task<bool> Login(string username, string password)
 
     public async Task<IEnumerable<SeriesInformation>> SearchSeries(string searchQuery)
     {
-        var searchHtml = await _httpClient.GetStringAsync($"{Constants.SdarotUrls.SearchUrl}{searchQuery}");
-        var doc = new HtmlDocument();
-        doc.LoadHtml(searchHtml);
+        var showsJson = await _httpClient.GetStringAsync($"{Constants.SdarotUrls.AllShowsUrl}");
+        var shows = JsonSerializer.Deserialize<List<SeriesInformation>>(showsJson);
+
+        var relevantShows = shows?.Where(x =>
+            x.SeriesNameHe.Contains(searchQuery, StringComparison.CurrentCultureIgnoreCase) ||
+            x.SeriesNameEn.Contains(searchQuery, StringComparison.CurrentCultureIgnoreCase)).ToList();
 
-        var seriesNameElement = doc.DocumentNode.SelectSingleNode(Constants.XPathSelectors.SeriesPageSeriesName);
+        // If none, turn to empty array.
+        relevantShows ??= new List<SeriesInformation>();
```

Comment:
Pretty sure `.ToList()` will already give you an empty list if the enumerable is empty, so this line is unnecessary